Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component } from '@angular/core';
+import { rrulestr } from 'rrule';
+import {CloseButtonOptions} from './interfaces';
+import {appointments, Appointment} from './data/appointments'
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css'],
+})
+
+export class AppComponent {
+  dataSource: Appointment[] = appointments;
+  currentDate: Date = new Date(2022, 9, 1);
+  popupVisible: boolean = false;
+  closeButtonOptions: CloseButtonOptions  = {
+    text: 'Close',
+    onClick: () => {
+      this.popupVisible = false;
+    }
+  };
+
+  checkDateInArray(dateString: string): number {
+    const weekDayNumbers = [6, 0, 1, 2, 3, 4, 5];
+    let date = new Date(dateString);
+    let day = date.getDay();
+
+    return weekDayNumbers[day];
+  }
+
+  handleAppointmentAdding(e: any): void {
```

Comment:
Unfortunately, ```onAppointmentAdding``` has type ```EventEmitter<any>```
But in our documentation, the event is described.
[Link to the documentation](https://js.devexpress.com/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/)

Let's replace ```any``` with the ```AppointmentAddingEvent``` type from devextreme package and rename ```e``` to ```event```
```suggestion
  handleAppointmentAdding(event: AppointmentAddingEvent): void {
```